test/stream_execution_test.cpp

```diff
@@ -0,0 +1,103 @@
+#include <migraphx/pre_scheduling.hpp>
+#include <migraphx/operators.hpp>
+#include <migraphx/generate.hpp>
+#include <migraphx/instruction.hpp>
+#include <migraphx/program.hpp>
+#include <basic_ops.hpp>
+#include <test.hpp>
+
+struct set_stream
+{
+    int stream = -1;
+    std::string name() const { return "gpu::set_stream"; }
```

Comment:
This shouldn't use the same name as the gpu so we can make sure we aren't hard-coding such names in our code.